pkg/controllers/cassandra/nodepool/nodepool.go

```diff
@@ -34,77 +36,65 @@ func NewControl(
 	}
 }
 
-func (e *defaultCassandraClusterNodepoolControl) removeUnusedStatefulSets(
+func (e *defaultCassandraClusterNodepoolControl) clusterStatefulSets(
 	cluster *v1alpha1.CassandraCluster,
-) error {
-	expectedStatefulSetNames := map[string]bool{}
-	for _, pool := range cluster.Spec.NodePools {
-		name := util.NodePoolResourceName(cluster, &pool)
-		expectedStatefulSetNames[name] = true
-	}
-	client := e.kubeClient.AppsV1beta1().StatefulSets(cluster.Namespace)
+) (results map[string]*v1beta1.StatefulSet, err error) {
+	results = map[string]*v1beta1.StatefulSet{}
```

Comment:
I'm confused what this package does now that we've switched to Actions for managing node pools? When I implemented this in the ES controller, the `nodepool` package was eventually refactored away entirely.

Reply:
I re-purposed it to be responsible only for updating the `CassandraCluster.Status.NodePools`.

Reply:
Given the above model of only deciding which action to perform next based on the Status field, can we move this into a `UpdateClusterStatus` function in the `cassandra` package? I like the idea of keeping this sort of info centralised, and it saves having the extra package (which given other similar packages e.g. `service` create resources, may be confusing in future).

*all* changes to the cluster status can then be computed and set in this function.

Reply:
I'm going to leave it like this for now.
My memory of previous conversations about the ``Controller.Sync`` pattern is that the ``Sync`` method is expected to mutate the ``CassandraCluster.Status`` and having run all the `control` s, the `Controller` performs an ``UpdateStatus`` with the final status.

Having it in a separate package makes it easier to unit test the NodePoolStatus behaviour in isolation.